xla/service/gpu/transforms/windowed_einsum_handler.cc

```diff
@@ -1186,6 +1178,44 @@ absl::StatusOr<bool> WindowedEinsumHandler::Run(
     changed |= visitor.changed();
   }
 
+  if (all_windowed_einsum_loops.size() > 0) {
+    TF_ASSIGN_OR_RETURN(bool applied_algsimp,
+                        AlgebraicSimplifier(AlgebraicSimplifierOptions())
+                            .Run(module, execution_threads));
+    changed |= applied_algsimp;
+    TF_ASSIGN_OR_RETURN(bool applied_cf,
+                        HloConstantFolding().Run(module, execution_threads));
```

Comment:
I'm somewhat nervous that these passes may free or invalid some instructions in `all_windowed_einsum_loops`. I don't think AlgebraicSimplifier can touch while-loop, but I'd rather not rely on this assumption. And maybe `HloConstantFolding` can get rid of while-loops.

Why do you need these passes? Please comment this. And can you run them before creating `all_windowed_einsum_loops`? Or alternatively, you can clear `all_windowed_einsum_loops` and find and put the while-loop instructions back in the vector after running the passes.

Reply:
Running this is to prepare the loops for unrolling, the unroller util function determines loop iterator with a very specific pattern which expects alg simp and constant fold applied. I checked that both passes wont touch while loops but i will document the intention to run these passes.